### PR TITLE
Move FAve 84H to data driven design + Via Custom UI

### PR DESCRIPTION
**Pull request by ToastyStoemp**
## Description

This updates the FAve 84 Hotswap board to data driven design.
Fixes the LED config that was removed due to incorrect configuration from previous PR.
Set the right flags on Mods / Alphas for LEDs

Add Custom UI VIA definitions for user control of the Caps lock indicator

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/linworks/fave84h/config.h [hunk 1]
```diff
@@ -14,49 +14,21 @@ GNU General Public License for more details.
...
+#define RGB_MATRIX_LED_COUNT 126
+#define RGB_DI_PIN D2
+
 #define RGB_DISABLE_WHEN_USB_SUSPENDED
 #define RGB_MATRIX_KEYPRESSES 
 #define RGB_MATRIX_FRAMEBUFFER_EFFECTS
 #define RGB_MATRIX_MAXIMUM_BRIGHTNESS 120
 
-#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL
+#define RGB_MATRIX_STARTUP_MODE RGB_MATRIX_CYCLE_ALL
```
Comment (zvecr) [inline keyboards/linworks/fave84h/config.h:31]: `RGB_MATRIX_STARTUP_MODE` isnt a thing anymore

```suggestion
#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL
```

**Review by zvecr — CHANGES_REQUESTED** [review]
You have a mixture of `_kb` functions defined at the keymap level. This is incorrect and needs to be changed to `_user` versions (should probably also use the basic `eeconfig_update_user` as you are only storing 4 bytes, which the basic functionality already provides ).

File: keyboards/linworks/fave84h/config.h (hunk 1 above)
Comment (waffle87) [inline keyboards/linworks/fave84h/config.h:31]:
```suggestion
#define RGB_MATRIX_DEFAULT_MODE RGB_MATRIX_CYCLE_ALL
```

File: keyboards/linworks/fave84h/config.h [hunk 2]
```diff
@@ -14,49 +14,21 @@ GNU General Public License for more details.
 You should have received a copy of the GNU General Public License
 along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
-
 #pragma once
 
 #include "config_common.h"
```
Comment (waffle87) [inline keyboards/linworks/fave84h/config.h:19]:
```suggestion
```

File: keyboards/linworks/fave84h/keymaps/via/keymap.c [hunk 3]
```diff
@@ -57,14 +57,140 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+void keyboard_post_init_kb(void) {
+    //Read our custom menu variables from memory
+    eeconfig_read_kb_datablock(&fave);
+}
+
+bool rgb_matrix_indicators_kb(void) {
+    // First decides if action is needed. If a user code is defined, or the indicator is disabled, then does not act.
+    if (!rgb_matrix_indicators_user()) {
+        return false;
+    }
```
Comment (waffle87) [inline keyboards/linworks/fave84h/keymaps/via/keymap.c:105]:
```suggestion
void keyboard_post_init_user(void) {
    //Read our custom menu variables from memory
    eeconfig_read_kb_datablock(&fave);
}

bool rgb_matrix_indicators_user(void) {
```

File: keyboards/linworks/fave84h/keymaps/via/keymap.c [hunk 4]
```diff
@@ -57,14 +57,140 @@ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
...
+} indicator_config;
+
+// Board memory layout
+typedef struct _fave_config_t {
+    indicator_config caps;
+} fave_config;
+
+fave_config fave;
+
+void eeconfig_init_kb(void) {
```
Comment (waffle87) [inline keyboards/linworks/fave84h/keymaps/via/keymap.c:78]:
```suggestion
void eeconfig_init_user(void) {
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

**Comment by ToastyStoemp** [comment]
> You have a mixture of `_kb` functions defined at the keymap level. This is incorrect and needs to be changed to `_user` versions (should probably also use the basic `eeconfig_update_user` as you are only storing 4 bytes, which the basic functionality already provides ).

Updated all Except for the `via_custom_value_command_kb` function due to the lack of a user version of that.

**Review by drashna — APPROVED** [review]

**Review by waffle87 — APPROVED** [review]

**Pull request merged**